repos/system_upgrade/el7toel8/files/rhel_upgrade.py

```diff
@@ -55,6 +55,34 @@ def pre_configure(self):
         self.base.conf.releasever = self.plugin_data['dnf_conf']['releasever']
 
     def configure(self):
+        def read_aws_region(repo):
+            region = None
+            if repo.baseurl:
+                region = repo.baseurl.split('.', 2)[1]
```

Comment:
@Skullman could you please take a look at this part? Is this way of getting the REGION fine?

Reply:
Yes,
standard format of AWS address is <machine>.REGION.<rest_of_domain_name> - both machine and rest of domain might differ a lot, but second field is always region string.